crcmocks/entitlements.py

```diff
@@ -1,28 +1,40 @@
 from flask import jsonify
 from flask import Blueprint
+from flask import request
+
+from crcmocks.config import DEFAULT_SERVICES
+from crcmocks.util import get_user_rh_identity
 
 
 blueprint = Blueprint("entitlements", __name__)
 
 
-services = [
-    "ansible",
-    "cost_management",
-    "insights",
-    "advisor",
-    "migrations",
-    "openshift",
-    "settings",
-    "smart_management",
-    "subscriptions",
-    "user_preferences",
-]
-entitlements = {}
-for svc in services:
-    entitlements[svc] = {"is_entitled": True, "is_trial": False}
-
-
-# TODO: allow customizable entitlements response per user
+# TODO: support is_trial?
 @blueprint.route("/v1/services")
 def services():
-    return jsonify(entitlements)
+    # get the identity header
+    identity_header = request.headers.get("X-Rh-Identity")
+
+    if not identity_header:
+        return "No identity header present in request\n", 401
+
+    # get the user
+    user, username, account_number = get_user_rh_identity(identity_header)
+
+    if not user:
+        return f"No user found in TinyDB with username:" \
+               f" {username} or account_number: {account_number}\n", 404
+    if len(user) > 1:
+        return f"Multiple users found with:" \
+               f" {username} or account_number: {account_number}\n", 406
+
+    # finally get the entitlements of the user
+    if user[0].get("entitlements") is not None:
+        entitled_services = user[0]["entitlements"].split(",")
+    else:
+        entitled_services = DEFAULT_SERVICES.copy()
+
+    entitlements = {}
+    for svc in DEFAULT_SERVICES:
```

Comment:
Why not loop on `entitled_services` here?

Reply:
The normal c.rh.c entitlements endpoint returns the not only services you're entitled to, but also services you're not entitled to.